client/src/main/java/io/pravega/client/tables/TableEntry.java

```diff
@@ -23,72 +24,110 @@
     /**
      * The {@link TableKey}.
      *
      * @return The content of {@link TableKey}.
      */
     @NonNull
-    private final TableKey<KeyT> key;
+    private final TableKey key;
 
     /**
      * The Value.
      *
      * @return The value associated to the {@link TableKey}.
      */
-    private final ValueT value;
+    private final ByteBuffer value;
+
+    /**
+     * Gets a value indicating whether this {@link TableEntry} has a specified {@link Version}.
+     *
+     * If versioned, then the following methods on {@link KeyValueTable} will be conditioned on this {@link TableEntry}'s
+     * {@link TableKey} matching the server-side version with that provided in this instance:
+     * - {@link KeyValueTable#put(TableEntry)}
+     * - {@link KeyValueTable#putAll(Iterable)}
+     *
+     * @return True if versioned, false otherwise.
+     */
+    public boolean isVersioned() {
+        return this.key.isVersioned();
+    }
+
+    /**
+     * Creates a new {@link TableEntry} with a Key made of only a Primary Key, with no specific version.
+     *
+     * @param primaryKey The Entry Key (Primary Key only).
+     * @param value      The Entry Value.
+     * @return An unversioned {@link TableEntry} (version set to {@link Version#NO_VERSION}).
+     */
+    public static TableEntry unversioned(ByteBuffer primaryKey, ByteBuffer value) {
+        return unversioned(primaryKey, null, value);
+    }
 
     /**
-     * Creates a new {@link TableEntry} with no specific version. When used with {@link KeyValueTable#replaceAll}, this
-     * {@link TableEntry} will be treated as an unconditional update.
+     * Creates a new {@link TableEntry} with a Key made of a Primary Key and Secondary Key, with no specific version.
      *
-     * @param key      The Entry Key.
-     * @param value    The Entry Value.
-     * @param <KeyT>   Key Type.
-     * @param <ValueT> Value Type.
+     * @param primaryKey   The Entry Key's Primary Key.
+     * @param secondaryKey The Entry Key's Secondary Key.
+     * @param value        The Entry Value.
      * @return An unversioned {@link TableEntry} (version set to {@link Version#NO_VERSION}).
      */
-    public static <KeyT, ValueT> TableEntry<KeyT, ValueT> unversioned(KeyT key, ValueT value) {
-        return new TableEntry<>(TableKey.unversioned(key), value);
+    public static TableEntry unversioned(ByteBuffer primaryKey, ByteBuffer secondaryKey, ByteBuffer value) {
+        return new TableEntry(TableKey.unversioned(primaryKey, secondaryKey), value);
+    }
+
+    /**
+     * Creates a new {@link TableEntry} with a Key made of only a Primary Key, with a version that indicates the key
+     * must not exist.
+     *
+     * @param primaryKey The Entry Key (Primary Key only).
+     * @param value      The Entry Value.
+     * @return A {@link TableEntry} with a version set to {@link Version#NOT_EXISTS}.
+     */
+    public static TableEntry notExists(ByteBuffer primaryKey, ByteBuffer value) {
```

Comment:
I find this function name confusing. IE: "How can an entry that does not exist have a value?"

Reply:
This is used for conditional inserts: condition the update on the predicate that the key must not exist. So: insert <key, value> if key does not already exist. 

If the Javadoc for this one isn't clear, I can change that - let me know.

Reply:
Would either `exclusive` or `absent` be better?

Reply:
I like `absent`. I'll take this one step further and rename `unversioned` to `anyVersion`. We'll have:
- `TableEntry.anyVersion`
- `TableEntry.versioned`
- `TableEntry.absent`

Reply:
I have removed all these methods in the API simplification I pushed on April 30.